### PR TITLE
Instance of else clause

### LINKED ISSUE #1719: Type guard with instanceof fails to narrow type in `else` block
Hi,

**TS Version**: 1.4

In the following:

```TypeScript
class Bar {
	bar ='bar';
}

class Foo {
 	foo = 100;
}

function f1(x: Bar|Foo) {
	
	if(x instanceof Bar){
		console.log(x.bar);
	}
	else {
		console.log(x.foo); // Error: inferred as Bar|Foo
	}	
}
 f1(new Bar()); // bar
 f1(new Foo()); // 100
```

is there any reason why in the `else` block we are failing to narrow the type to `Foo`?

### PR DESCRIPTION
**Pull request by tinganho**
Fixes #1719.

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -5976,6 +5977,13 @@ namespace ts {
                 }
 
                 if (targetType) {
+                    if (!assumeTrue) {
+                        if (type.flags & TypeFlags.Union) {
+                            return getUnionType(filter((<UnionType>type).types, t => !isTypeSubtypeOf(t, targetType)));
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: @RyanCavanaugh what happened with the idea that if the RHS type isn't a subtype of any types in the union, that we just discard the LHS type and take the RHS type? Is that something that we eventually went with?

Comment (RyanCavanaugh) [inline src/compiler/checker.ts]: I don't remember what we decided on that.

**Comment by RyanCavanaugh** [comment]
@tinganho can you address the merge conflict?

**Comment by RyanCavanaugh** [comment]
What happened to the `tsxAttributeResolution9` baselines?

**Comment by tinganho** [comment]
@RyanCavanaugh I fixed the merge conflict. And `tsxAttributeResolution9` doesn't appear as deleted anymore when I rebased my solution.

**Comment by RyanCavanaugh** [comment]
@mhegazy good with this?

**Comment by DanielRosenwasser** [comment]
:+1:

**Comment by mhegazy** [comment]
Sorry for the delay. :+1:

**Comment by mhegazy** [comment]
thanks!

**Comment by tinganho** [comment]
:+1:

**Pull request merged**